databases/week3/week3.sql

```diff
@@ -0,0 +1,166 @@
+
+-- Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('3', '2', '2020-06-01');
+
+-- Get a reservation with any id, fx 1
+SELECT * FROM reservation
+WHERE id = 1;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation SET  number_of_guests = '4'
+WHERE id=1; 
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM reservation 
+WHERE id = 1;
+
+						-- TABLE REVIEW--
+-- Get all reviews
+SELECT * FROM review;  
+
+-- Add a new review  
+INSERT INTO review (title, description, meal_id, stasr, created_date )
+VALUES ('perfect', 'meal was amaizing', '2', '5', '2020-06-01');                    
+
+-- Get a review with any id, fx 1
+SELECT * FROM review
+WHERE id = 1;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review SET title = 'good'
+WHERE id=1; 
+
+-- Delete a review with any id, fx 1
+DELETE FROM review
+WHERE id = 1;
+
+-- Additional queries
+/*Now add a couple of different meals, reservations and reviews with different attributes. With those meals create the following queries*/
+
+						-- TABLE REVIEW--
+INSERT INTO review (title, description, meal_id, stars, created_date )
+VALUES ('amazing', 'super cervice', '2', '5', '2020-06-01');
+INSERT INTO review (title, description, meal_id, stars, created_date )
+VALUES ('bad', 'too expensive', '5', '1', '2020-06-06');
+INSERT INTO review (title, description, meal_id, stars, created_date )
+VALUES ('great', 'I will come again', '3', '3', '2020-06-05');
+
+						-- TABLE RESERVATION --
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('2', '3', '2020-06-09');
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('1', '4', '2020-06-03');
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('2', '3', '2020-06-07');
+
+						-- TABLE MEAL --
+INSERT INTO meal ( title, description, location, `when`, max_reservation, price, created_date)
+VALUES ('supe', 'with meat bolls', 'south', '2020-06-8', '2', 20.50, '2020-06-04');
+INSERT INTO meal ( title, description, location, `when`, max_reservation, price, created_date)
+VALUES ('spaghetti', 'with  basilic', 'west', '2020-06-10', '3', 45, '2020-06-04');
+INSERT INTO meal ( title, description, location, `when`, max_reservation, price, created_date)
+VALUES ('ice cream', 'organic', 'city center', '2020-06-15', '5', 45, '2020-06-04');
+
+-- Functionality
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT title FROM meal
+WHERE price < 50;
+
+-- Get meals that still has available reservations
+select * from (SELECT meal.title, meal.max_reservation AS max_reservations, COUNT(reservation.id) AS reserved_meals_count
+FROM reservation
+LEFT JOIN meal ON meal.id = reservation.meal_id
+GROUP BY meal.title) AS meals_reservations
+WHERE max_reservations > reserved_meals_count;
```

Comment:
I'm getting this error: `Error Code: 1055. Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mealsharing.meal.max_reservation' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by`

Also: While it is sometimes necessary to use a subquery (a select inside another select), it's not in this case. Try to see if you can do the query without the subquery.

Reply:
I fixed the error. but I could not remove a subquery. in my case because of COUNT. Honestly, the task wasn't clear to me.  Should I base on the amount of meal reservation id (table reservation) or on the amount of 'number_of_guests'? I chose 'reservation_id' , in my opinion, I should be worried about how many dishes will be ordered and no how many people will eat them. I would appreciate your help.